File: Actito.iOS.Binding.Bridge/Sources/Models/ActitoDevice.swift
import Foundation
import ActitoKit

@objc
public class ActitoDevice : NSObject {
    @objc public let deviceId: String
    @objc public let userId: String?
    @objc public let userName: String?
    @objc public let timeZoneOffset: Float
    @objc public let dnd: ActitoDoNotDisturb?
    @objc public let userData: [String : String]
    @objc public let backgroundAppRefresh: Bool

    @objc public init(deviceId: String, userId: String?, userName: String?, timeZoneOffset: Float, dnd: ActitoDoNotDisturb?, userData: [String : String], backgroundAppRefresh: Bool) {
        self.deviceId = deviceId
        self.userId = userId
        self.userName = userName
        self.timeZoneOffset = timeZoneOffset
        self.dnd = dnd
        self.userData = userData
        self.backgroundAppRefresh = backgroundAppRefresh
    }

    public convenience init(from device: ActitoKit.ActitoDevice) {
        self.init(
            deviceId: device.id,
            userId: device.userId,
            userName: device.userName,
            timeZoneOffset: device.timeZoneOffset,
            dnd: device.dnd.map { ActitoDoNotDisturb(from: $0) },
            userData: device.userData,
            backgroundAppRefresh: device.backgroundAppRefresh
        )
    }
}

File: Actito.Core/Models/ActitoDevice.cs
namespace ActitoSdk.Core.Models;

/// <summary>
/// Represents a device registered in Actito.
/// </summary>
/// <remarks>
/// An <see cref="ActitoDevice"/> is associated with a physical device and may be optionally
/// linked to a user. It contains timezone information, user-related metadata,
/// and optional configuration such as do-not-disturb settings.
/// </remarks>
public class ActitoDevice
{
    /// <summary>
    /// Unique identifier of the device.
    /// </summary>
    public string Id { get; }

    /// <summary>
    /// Optional identifier of the user associated with the device.
    /// </summary>
    public string? UserId { get; }

    /// <summary>
    /// Optional display name of the associated user.
    /// </summary>
    public string? UserName { get; }

    /// <summary>
    /// ime zone offset of the device in hours relative to UTC.
    /// </summary>
    public double TimeZoneOffset { get; }

    /// <summary>
    /// Optional <see cref="ActitoDoNotDisturb"/> configuration for the device.
    /// </summary>
    public ActitoDoNotDisturb? Dnd { get; }

    /// <summary>
    /// Custom user data associated with the device.
    /// </summary>
    /// <remarks>
    /// This map contains key–value pairs representing user attributes or profile
    /// information linked to the device.
    /// </remarks>
    public IDictionary<string, string> UserData { get; }

    /// <summary>
    /// Constructor for <see cref="ActitoDevice"/>.
    /// </summary>
    public ActitoDevice(string id, string? userId, string? userName, double timeZoneOffset, ActitoDoNotDisturb? dnd, IDictionary<string, string> userData)
    {
        Id = id;
        UserId = userId;
        UserName = userName;
        TimeZoneOffset = timeZoneOffset;
        Dnd = dnd;
        UserData = userData;
    }
}
